Publish Roblox.Launcher (v1.6.1.31805) using official source(s)

thomasthebro1 opened the pull request:
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/17877)

thomasthebro1 committed:
feat(launcher): Publish Roblox.Launcher (v1.6.0.31052) using official source(s).

KaranKad commented on manifests/r/Roblox/Launcher/1.6.0.31052/Roblox.Launcher.installer.yaml:
```diff
@@ -0,0 +1,17 @@
+Installers:
+  - Architecture: x86
+    InstallerType: nullsoft
+    InstallerSwitches:
+      Silent: /S
+      SilentWithProgress: /S
```
```suggestion
```

KaranKad replied:
I forgot to tell that you can remove these lines because `InstallerType` is set to `nullsoft`

thomasthebro1 replied:
I'll sort that now. Thanks for the heads up!

thomasthebro1 commented:
Looking into this now

thomasthebro1 committed:
fix(launcher): Fix URLs? (maybe)

thomasthebro1 committed:
More fixes

KaranKad commented:
If it doesn't work this time try using direct link `https://setup.rbxcdn.com/version-7d96d7dad25d49f1-Roblox.exe`

thomasthebro1 commented:
It's not flagging the `InstallerUrl` as a bad URL it's the `PrivacyUrl` and I am not exactly sure why. I have put it in quotes to see if it will work this time.

thomasthebro1 commented:
Here we go again haha

thomasthebro1 committed:
fix: Temporarily comment out PrivacyUrl

thomasthebro1 commented:
Well at least we are progressing further now

KaranKad commented:
![image](https://user-images.githubusercontent.com/71691514/122442972-befd3a80-cfbc-11eb-872b-cd20bcd5368a.png)
It's unable to download for some reason.

thomasthebro1 commented:
Yes I am aware about this I am not exactly sure why this happens but sometimes it works and then other times it doesn't?

thomasthebro1 commented:
> Hello @thomasthebro1,
> The package manager bot determined there was an issue with installing the application correctly. Please check the application installs correctly. Once repaired, please push an update to your pull request.

Does anybody know why this might be happening?

KaranKad commented:
> When accessing the URL directly it works fine so could it possibly be something to do with winget-cli?

This might be winget-cli related

OfficialEsco commented:
https://redirectdetective.com/
The `PrivacyUrl` is `403: Forbidden` (Will fail in most bot cases)
The `InstallerUrl` `https://setup.rbxcdn.com/version-7d96d7dad25d49f1-Roblox.exe` is just a temporary link so `https://www.roblox.com/download/client` is the best to use
The `Validation-Installation-Error` might be wingetbot being childish, try closing and reopening the case.

thomasthebro1 commented:
Ok I'll try closing and reopening the PR.

thomasthebro1 committed:
feat(launcher): Update to v1.6.1.31805 and use installer binary link

thomasthebro1 commented:
Investigating

thomasthebro1 committed:
fix: Update and fix SHA256

thomasthebro1 commented:
I've just encountered a rare sighting, please for the love of god work this time 🙏🏼 
![image](https://user-images.githubusercontent.com/43547293/122449357-ae899780-cf9d-11eb-9d80-d2d83be7f980.png)

thomasthebro1 committed:
fix: Try using exe (wingetcreate suggested)

KaranKad commented:
The application installs itself but leaves this popup behind and winget thinks the application is still installing. Unless someone clicks `ok` winget will think application is not installed, hence `Validation-Unattended-Failed`.
Also when I changed winget's network settings to following, it does not give me error `0x80072f76 : unknown error` every time.
```
"network": {
        "downloader": "wininet"
    } 
```
![image](https://user-images.githubusercontent.com/71691514/122457868-c298bd80-cfcc-11eb-9bb9-42ea3071665c.png)

thomasthebro1 commented:
Interesting I will look into this now. This might be why Validation is taking forever

thomasthebro1 commented:
How would I go around fixing this problem? @KaranKad

KaranKad commented:
I don't think winget-cli can currently do anything about it, so we might have to wait on this package.

OfficialEsco commented:
I don't even think the silent switch does anything, you might have to find a switch for nolaunch or something along those lines.
Or you have to wait for support for not 100% silent

denelon commented:
We are getting ready to build a new release with https://github.com/microsoft/winget-cli/pull/1138, so that should help with the DO challenges.

thomasthebro1 commented:
In the meantime while a fix is found for this package, I am working on contributing to a different one, keep a look out :)

thomasthebro1 committed:
Merge branch 'microsoft:master' into master

thomasthebro1 committed:
delete(launcher): Remove Roblox Launcher

Remove the Roblox launcher until I can find a fix that auto-completes the installation without having to press OK at the end.

thomasthebro1 commented:
I'm going to close this PR until I can find a fix which completes the installation at the end without pressing OK. winget-cli might require a feature suggestion as I think the Roblox Launcher uses a custom installer

rollingmoai commented:
@thomasthebro1 You might be interested with [Bloxstrap](https://github.com/pizzaboxer/bloxstrap), which is available in Winget.
